### ISSUE TITLE
[JENKINS-9598] Overhaul bundled plugins or setup experience

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#4479]
<p>It would be useful to include support for Git in the standard jenkins distribution. </p>

<p>As a new Jenkins user, I will like to choose a feature area that I am interested in and Jenkins installs the related plugins for me.<br/>
As an existing Jenkins user who is upgrading, I will like to choose the feature areas that I am interested in and Jenkins installs the related plugins for me.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/wakaleo.png" title="wakaleo's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wakaleo">wakaleo</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598" target="_blank">Overhaul bundled plugins or setup experience</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.0, setup, ui, user-experience
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-01-26T12:44:40+00:00
<li><b>votes</b>: 24
<li><b>watchers</b>: 20
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>It would be useful to include support for Git in the standard jenkins distribution. </p>

<p>As a new Jenkins user, I will like to choose a feature area that I am interested in and Jenkins installs the related plugins for me.
As an existing Jenkins user who is upgrading, I will like to choose the feature areas that I am interested in and Jenkins installs the related plugins for me.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kreyssel.png" title="kreyssel's avatar" /> kreyssel:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=148391&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148391">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1</p>
</pre>
</details>

</ul>
</details>

<p>+1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/timp.jpg" title="timp's avatar" /> timp:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=148404&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148404">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1</p>
</pre>
</details>

</ul>
</details>

<p>+1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/wakaleo.png" title="wakaleo's avatar" /> wakaleo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=148618&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148618">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The idea is to get users up and running as quickly as possible. (The other idea is so that readers of the 'First Steps' chapter in 'Jenkins: The Definitive Guide' can get a few builds running with a minimum of configuration <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> ).</p>

<p>Seriously, though, I'd even be prone to bundle recent versions of Maven (e.g. 2.2.1 and 3.0.x) and Ant for the same reasons. Beyond that, the idea of themed plugin sets does sound quite cool.</p>
</pre>
</details>

</ul>
</details>

<p>The idea is to get users up and running as quickly as possible. (The other idea is so that readers of the 'First Steps' chapter in 'Jenkins: The Definitive Guide' can get a few builds running with a minimum of configuration <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> ).</p>

<p>Seriously, though, I'd even be prone to bundle recent versions of Maven (e.g. 2.2.1 and 3.0.x) and Ant for the same reasons. Beyond that, the idea of themed plugin sets does sound quite cool.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kreyssel.png" title="kreyssel's avatar" /> kreyssel:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=148638&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148638">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@aheretier
Is its relevant to have SVN/CVS plugin in distribution? Today, i think git is relevant.</p>
</pre>
</details>

</ul>
</details>

<p>@​aheretier<br/>
Is its relevant to have SVN/CVS plugin in distribution? Today, i think git is relevant.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cowwoc.jpg" title="cowwoc's avatar" /> cowwoc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=168373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-168373">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I agree with David. Improve the UI, as opposed to bundling unnecessary functionality into core.</p>
</pre>
</details>

</ul>
</details>

<p>I agree with David. Improve the UI, as opposed to bundling unnecessary functionality into core.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=202258&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202258">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I suppose that several "bundle plugins" would be enough (with a separate page on the update center?).
Any plugin could just reference other plugins, but it also could provide setup dialogs or initialize the configuration.</p>
</pre>
</details>

</ul>
</details>

<p>I suppose that several "bundle plugins" would be enough (with a separate page on the update center?).<br/>
Any plugin could just reference other plugins, but it also could provide setup dialogs or initialize the configuration.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=202361&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202361">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Having gone down the road of “bundle plugins” in another context (NetBeans), I can say that they cause more trouble and confusion than they solve.</p>
</pre>
</details>

</ul>
</details>

<p>Having gone down the road of “bundle plugins” in another context (NetBeans), I can say that they cause more trouble and confusion than they solve.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=202435&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202435">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As I mentioned on the mailing list, the first run experience is kind of weird. A guided tour through the basic concepts, the most important settings of <em>Configure System</em>, and a subset of <em>Manage Plugins</em> (as to not overwhelm the newcomer) might be helpful.</p>

<p>IMO the only bundled plugins should be whatever used to be in core, plus their new dependencies, for compatibility reasons. And those might not even need installing by default, but only if an older version that might otherwise break (e.g. &lt;1.466 for LDAP plugin) is detected.</p>
</pre>
</details>

</ul>
</details>

<p>As I mentioned on the mailing list, the first run experience is kind of weird. A guided tour through the basic concepts, the most important settings of <em>Configure System</em>, and a subset of <em>Manage Plugins</em> (as to not overwhelm the newcomer) might be helpful.</p>

<p>IMO the only bundled plugins should be whatever used to be in core, plus their new dependencies, for compatibility reasons. And those might not even need installing by default, but only if an older version that might otherwise break (e.g. <1.466 for LDAP plugin) is detected.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-9598?focusedId=237576&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-237576">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: kzantow
Path:
 core/src/main/java/hudson/model/UpdateCenter.java
 core/src/main/java/jenkins/I18n.java
 core/src/main/java/jenkins/model/Jenkins.java
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties
 core/src/main/resources/jenkins/install/pluginSetupWizard_de.properties
 core/src/main/resources/jenkins/install/pluginSetupWizard_es.properties
 core/src/main/resources/lib/layout/layout.jelly
 war/gulpfile.js
 war/package.json
 war/pom.xml
 war/src/main/js/pluginManager.js
 war/src/main/js/pluginSetupWizard.js
 war/src/main/js/pluginSetupWizardGui.js
 war/src/main/js/recommendedPlugins.js
 war/src/main/js/templates/errorPanel.hbs
 war/src/main/js/templates/offlinePanel.hbs
 war/src/main/js/templates/pluginSelectionPanel.hbs
 war/src/main/js/templates/pluginSetupWizard.hbs
 war/src/main/js/templates/progressPanel.hbs
 war/src/main/js/templates/successPanel.hbs
 war/src/main/js/templates/welcomePanel.hbs
 war/src/main/js/templates/wizard.hbs
 war/src/main/js/util/ajax.js
 war/src/main/js/util/browser.js
 war/src/main/js/util/jenkins.js
 war/src/main/js/util/system.js
 war/src/main/js/wizard.js
 war/src/main/less/pluginSetupWizard.less
 war/src/main/less/wizard.less
 war/src/main/webapp/WEB-INF/web.xml
 war/src/test/js/pluginManager-spec.js
 war/src/test/js/pluginSetupWizard-spec.js
 war/src/test/js/util.js
 war/src/test/js/wizard-spec.js
<a href="http://jenkins-ci.org/commit/jenkins/0b0c7f7bd9b52e16fbf4d372e366b9844c8f2f41" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/0b0c7f7bd9b52e16fbf4d372e366b9844c8f2f41</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-9598" title="Overhaul bundled plugins or setup experience" class="issue-link" data-issue-key="JENKINS-9598"><del>JENKINS-9598</del></a> - Initial installation wizard &amp; curated list of plugins</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: kzantow<br/>
Path:<br/>
 core/src/main/java/hudson/model/UpdateCenter.java<br/>
 core/src/main/java/jenkins/I18n.java<br/>
 core/src/main/java/jenkins/model/Jenkins.java<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard_de.properties<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard_es.properties<br/>
 core/src/main/resources/lib/layout/layout.jelly<br/>
 war/gulpfile.js<br/>
 war/package.json<br/>
 war/pom.xml<br/>
 war/src/main/js/pluginManager.js<br/>
 war/src/main/js/pluginSetupWizard.js<br/>
 war/src/main/js/pluginSetupWizardGui.js<br/>
 war/src/main/js/recommendedPlugins.js<br/>
 war/src/main/js/templates/errorPanel.hbs<br/>
 war/src/main/js/templates/offlinePanel.hbs<br/>
 war/src/main/js/templates/pluginSelectionPanel.hbs<br/>
 war/src/main/js/templates/pluginSetupWizard.hbs<br/>
 war/src/main/js/templates/progressPanel.hbs<br/>
 war/src/main/js/templates/successPanel.hbs<br/>
 war/src/main/js/templates/welcomePanel.hbs<br/>
 war/src/main/js/templates/wizard.hbs<br/>
 war/src/main/js/util/ajax.js<br/>
 war/src/main/js/util/browser.js<br/>
 war/src/main/js/util/jenkins.js<br/>
 war/src/main/js/util/system.js<br/>
 war/src/main/js/wizard.js<br/>
 war/src/main/less/pluginSetupWizard.less<br/>
 war/src/main/less/wizard.less<br/>
 war/src/main/webapp/WEB-INF/web.xml<br/>
 war/src/test/js/pluginManager-spec.js<br/>
 war/src/test/js/pluginSetupWizard-spec.js<br/>
 war/src/test/js/util.js<br/>
 war/src/test/js/wizard-spec.js<br/>
<a href="http://jenkins-ci.org/commit/jenkins/0b0c7f7bd9b52e16fbf4d372e366b9844c8f2f41" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/0b0c7f7bd9b52e16fbf4d372e366b9844c8f2f41</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-9598" title="Overhaul bundled plugins or setup experience" class="issue-link" data-issue-key="JENKINS-9598"><del>JENKINS-9598</del></a> - Initial installation wizard & curated list of plugins</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-23/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-20617%22">JENKINS-20617</a>]</i>
